data/domain/world-6/sneaking.tsx

```diff
@@ -204,4 +205,15 @@ export class Sneaking extends Domain {
             upgrade.costExponent = serverVars["A_empoExpon"];
         })
     }
+}
+
+export const updateSneaking = (data: Map<string, any>) => {
+    const alchemy = data.get("alchemy") as Alchemy;
+    const sneaking = data.get("sneaking") as Sneaking;
+
+    // The bubble say that the exponent part is reduced, but that's actually just a standard price reduction from what I calculated
+    const currencyConduitUpgrade = sneaking.sneakingUpgrades.find(upgrade => upgrade.index == 8);
```

Comment:
So ... I had a look at the code and yeah .. this is far from over 😆 

Wall of text coming:
```js
if ("JadeUpgCost" == e)
    return 8 == i
      ? (1 / (1 + q._customBlock_Ninja("NinjaBonus", 11, -1) / 100)) *
          (1 /
            (1 +
              c.asNumber(a.engine.getGameAttribute("DNSM").h.AlchBubbles.h.Y10) /
                100)) *
          c.asNumber(a.engine.getGameAttribute("CustomLists").h.NinjaUpg[0 | i][7]) *
          Math.pow(
            c.asNumber(a.engine.getGameAttribute("CustomLists").h.NinjaUpg[0 | i][8]),
            c.asNumber(a.engine.getGameAttribute("Ninja")[103][0 | i])
          )
      : (1 / (1 + q._customBlock_Ninja("NinjaBonus", 11, -1) / 100)) *
          c.asNumber(a.engine.getGameAttribute("CustomLists").h.NinjaUpg[0 | i][7]) *
          Math.pow(
            c.asNumber(a.engine.getGameAttribute("CustomLists").h.NinjaUpg[0 | i][8]),
            c.asNumber(a.engine.getGameAttribute("Ninja")[103][0 | i])
          );
```

So this is the code for the cost, ignoring the fact it's called `JadeUpgCost` which is very confusing considering the emporium. 

For currency conduit it looks like the math is:
`1 / (1 + discountFromNinjaItem11 / 100) * (1 / 1 + bubbleDiscount / 100) * costBase * Math.pow(costExponent, level))`

Others are the same just without the bubble discount.

So from that:
1. I don't think your cost multiplier is correct since (assuming 30% discount) -> (1 / 1.3) != (1 - 0.3) although they are pretty close.
2. You aren't accounting for the 11th ninja trinket being in the inventory (the name is `Gold Coupon` if you want to search for it and laugh at the description text that has a word not used anywhere else)

So yeah, few more things to go 😆

Reply:
Oh '-'

Gonna implement the calcul like that then

But in my defense, discounts are usually calculated by multiplying the original value by a number under 1
0.70 for a 30% discount, as 0.3 IS 30% of 1, while 1/3 isn't 

But I guess Lava made the game so he calculates as he wish x)